2.4/Dockerfile

```diff
@@ -0,0 +1,36 @@
+FROM centos:centos7
+
+# MongoDB image for OpenShift.
+#
+# Volumes:
+#  * /var/lib/mongodb/ - Datastore for MongoDB
+# Environment:
+#  * $MONGODB_USER - Database user name
+#  * $MONGODB_PASSWORD - User's password
+#  * $MONGODB_DATABASE - Name of the database to create
+
+# Image metadata
+ENV MONGODB_VERSION         2.4  
+ENV IMAGE_DESCRIPTION       MongoDB 2.4
+ENV IMAGE_TAGS              mongodb,mongodb24
+ENV IMAGE_EXPOSE_SERVICES   27017:mongodb
+
+MAINTAINER  Jakub Hadvig <jhadvig@redhat.com>
+EXPOSE 27017
+
+RUN yum install -y epel-release && \
+    rpmkeys --import file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7 && \
+    yum -y --setopt=tsflags=nodocs install \
+    https://www.softwarecollections.org/en/scls/rhscl/v8314/epel-7-x86_64/download/rhscl-v8314-epel-7-x86_64.noarch.rpm \
+    https://www.softwarecollections.org/en/scls/rhscl/mongodb24/epel-7-x86_64/download/rhscl-mongodb24-epel-7-x86_64.noarch.rpm && \
+    yum install -y --setopt=tsflags=nodocs v8314 mongodb24-mongodb mongodb24 && \
+    yum clean all
+
+COPY contrib/etc/ /opt/openshift/etc
```

Comment:
Since @mnagy already set some conventions around it, I'd prefer that dir be called `mongo`, see https://github.com/openshift/mysql/blob/master/5.5/Dockerfile#L29